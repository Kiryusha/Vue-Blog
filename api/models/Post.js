'use strict';
var mongoose = require('mongoose');
var Schema = mongoose.Schema;

var PostSchema = new Schema({
  title: String,
  code: String,
  date: {
    type: Date,
    default: Date.now
  },
  category: String,
  previewText: String,
  detailText: String,
});

module.exports = mongoose.model('Post', PostSchema);
