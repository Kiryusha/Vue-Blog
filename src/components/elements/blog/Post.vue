<template lang="pug">
  .post
    .post__heading
      .post__title Lorem ipsum dolor sit amet, consectetur adipiscing elit
      .post__info 22 апреля 2018, Кир
    .post__text
      p
        | Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
        | tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
        | veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
        | commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
        | velit esse cillum dolore eu fugiat nulla pariatur.
    .post__controls
      Button(
        :view="'more'"
      ) Подробнее
</template>

<script>
import Button from '../general/Button';

export default {
  components: {
    Button,
  },
};
</script>

<style lang="stylus">
  .post

    &__title
      font-size 30px

    &__info
      font-size 16px
      color #CF
      padding-top 10px

    &__text
      padding-top 10px
      font-size 16px
      line-height 1.7

    &__controls
      padding-top 20px
      display flex
      justify-content flex-end
</style>
