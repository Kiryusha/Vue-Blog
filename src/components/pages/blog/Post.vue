<template lang="pug">
  section.post
    .post__content
      .post__title-block
        h1.post__title {{ post.title }}
        .post__info {{ post.info }}
      .post__text-block(
        v-html="post.detailText"
      )
      .post__author
</template>

<script>
import axios from 'axios';

export default {
  data() {
    return {
      loading: false,
      post: {},
    };
  },
  mounted() {
    this.fetchData(this.$route.params.id);
  },
  beforeRouteUpdate(to, from, next) {
    this.fetchData(to.params.id);
    next();
  },
  methods: {
    fetchData() {
      this.$Progress.start();

      axios.get('/static/data/post.json').then((response) => {
        this.post = response.data;
        this.$Progress.finish();
      });
    },
  },
};
</script>

<style lang="stylus">
  .post
    card()

    &__title-block
      margin-bottom 70px

    &__title
      font-family 'Roboto Slab', sans-serif
      display inline-block
      margin 0
      font-size 34px
      font-weight normal
      margin-bottom 8px

    &__info
      font-family 'Roboto Slab', sans-serif
      font-size 18px
      color #c

    &__text-block
      font-size 18px
      line-height 1.5

      h1, h2, h3, h4, h5
        font-family 'Roboto Slab', sans-serif

      a
        text-decoration none
        color $yellow
        border-bottom 1px solid $yellow
        transition .2s

        &:hover
          color $yellowDark
          border-color $yellowDark

      cite
        display inline-block
        padding 20px 0 20px 40px
        color #9

      img
        width 100%

</style>
