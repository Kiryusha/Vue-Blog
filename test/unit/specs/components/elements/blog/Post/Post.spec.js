// Post.vue

import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Post from 'Components/elements/blog/Post';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Blog.vue', () => {
  const config = {
    localVue,
    mocks: {
      $route: {
        params: {
          code: 'test',
        }
      }
    }
  };

  let storeConfig = {
    state: {
      post: {
        data: {
          title: 'test',
          date: new Date('01/01/01'),
        },
      },
    },
    actions: {
      fetchPost: () => {},
    }
  };
  let store = new Vuex.Store(storeConfig);

  describe('component shallow rendering', () => {
    it('should match a snapshot - not authenticated', () => {
      const wrapper = shallowMount(Post, { ...config, store });
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('lifecycle-driven methods calls', () => {
    it('should set author to user name on created', () => {
      storeConfig.state.post.data.username = 'test';
      let store = new Vuex.Store(storeConfig);
      const wrapper = shallowMount(Post, { ...config, store });

      expect(wrapper.vm.author).toBe('test');
    });
  });
});
