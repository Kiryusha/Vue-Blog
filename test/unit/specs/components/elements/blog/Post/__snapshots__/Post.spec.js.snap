// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog.vue component shallow rendering should match a snapshot - not authenticated 1`] = `
<section class="post _loaded">
  <div class="content">
    <div class="title-block">
      <h1 class="title">test</h1>
      <div class="info">01 Января 2001, без автора</div>
    </div>
    <div class="text-block"></div>
  </div>
</section>
`;
